cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaStatusTest.java

```diff
@@ -957,6 +957,100 @@ public void testInitialStatusOnOldResource() throws ParseException {
+        Kafka oldKafka = new KafkaBuilder(getKafkaCrd())
+                .editOrNewSpec()
+                    .editOrNewKafka()
+                        .withNewPersistentClaimStorage()
+                            .withNewSize("100Gi")
+                        .endPersistentClaimStorage()
+                    .endKafka()
+                .endSpec()
+                .build();
+        KafkaCluster kafkaCluster = KafkaCluster.fromCrd(oldKafka, VERSIONS);
+
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(false);
+
+        // Mock the CRD Operator for Kafka resources
+        CrdOperator mockKafkaOps = supplier.kafkaOperator;
+        when(mockKafkaOps.getAsync(eq(namespace), eq(clusterName))).thenReturn(Future.succeededFuture(kafka));
+
+        ArgumentCaptor<Kafka> kafkaCaptor = ArgumentCaptor.forClass(Kafka.class);
+        when(mockKafkaOps.updateStatusAsync(kafkaCaptor.capture())).thenReturn(Future.succeededFuture());
+
+        // Mock the KafkaSetOperator
+        KafkaSetOperator mockKafkaSetOps = supplier.kafkaSetOperations;
+        when(mockKafkaSetOps.getAsync(eq(namespace), eq(KafkaCluster.kafkaClusterName(clusterName)))).thenReturn(Future.succeededFuture(kafkaCluster.generateStatefulSet(false, null, null)));
+
+        // Mock the ConfigMapOperator
+        ConfigMapOperator mockCmOps = supplier.configMapOperations;
+        when(mockCmOps.getAsync(eq(namespace), eq(clusterName))).thenReturn(Future.succeededFuture(kafkaCluster.generateMetricsAndLogConfigMap(null)));
+
+        // Mock Pods Operator
+        /*Pod pod0 = new PodBuilder()
+                .withNewMetadata()
+                .withNewName(clusterName + "-kafka-" + 0)
+                .endMetadata()
+                .withNewStatus()
+                .withNewHostIP("10.0.0.1")
+                .endStatus()
+                .build();
+
+        Pod pod1 = new PodBuilder()
+                .withNewMetadata()
+                .withNewName(clusterName + "-kafka-" + 1)
+                .endMetadata()
+                .withNewStatus()
+                .withNewHostIP("10.0.0.1")
+                .endStatus()
+                .build();
+
+        Pod pod2 = new PodBuilder()
+                .withNewMetadata()
+                .withNewName(clusterName + "-kafka-" + 2)
+                .endMetadata()
+                .withNewStatus()
+                .withNewHostIP("10.0.0.1")
+                .endStatus()
+                .build();
+
+        List<Pod> pods = new ArrayList<>();
+        pods.add(pod0);
+        pods.add(pod1);
+        pods.add(pod2);
+
+        PodOperator mockPodOps = supplier.podOperations;
+        when(mockPodOps.listAsync(eq(namespace), any(Labels.class))).thenReturn(Future.succeededFuture(pods));
+
+        // Mock Node operator
+        NodeOperator mockNodeOps = supplier.nodeOperator;
+        when(mockNodeOps.listAsync(any(Labels.class))).thenReturn(Future.succeededFuture(getClusterNodes()));*/
```

Comment:
what about this big commented code block?

Reply:
Good point. Thanks.